packages/models-library/src/models_library/service_settings.py

```diff
@@ -59,11 +66,145 @@ class Config:
         }
 
 
-class ServiceSettings(BaseModel):
-    __root__: List[ServiceSetting]
+class SimcoreServiceSettings(BaseModel):
+    __root__: List[SimcoreServiceSetting]
 
     def __iter__(self):
         return iter(self.__root__)
 
     def __getitem__(self, item):
         return self.__root__[item]
+
+
+class PathsMapping(BaseModel):
+    inputs_path: Path = Field(
+        ..., description="path where the service expects all the inputs folder"
+    )
+    outputs_path: Path = Field(
+        ..., description="path where the service expects all the outputs folder"
+    )
+    other_paths: List[Path] = Field(
+        [],
+        description="optional list of path which contents need to be saved and restored",
+    )
+
+    @validator("other_paths", always=True)
+    @classmethod
+    def convert_none_to_empty_list(cls, v):
+        return [] if v is None else v
+
+    class Config(_BaseConfig):
+        schema_extra = {
+            "examples": {
+                "outputs_path": "/tmp/outputs",  # nosec
+                "inputs_path": "/tmp/inputs",  # nosec
+            }
+        }
+
+
+ComposeSpecModel = Optional[Dict[str, Any]]
+
+
+class SimcoreService(BaseModel):
+    """Validate all the simcores.services.* labels on a service"""
+
+    settings: Json[SimcoreServiceSettings] = Field(
+        ...,
+        alias="simcore.service.settings",
+        description=(
+            "Contains setting like environment variables and "
+            "resource constraints which are required by the service"
+        ),
+    )
+
+    paths_mapping: Json[Optional[PathsMapping]] = Field(
+        None,
+        alias="simcore.service.paths-mapping",
+        description="json encoded, determines where the outputs and inputs directories are",
+    )
+
+    compose_spec: Json[ComposeSpecModel] = Field(
+        None,
+        alias="simcore.service.compose-spec",
+        description="json encoded docker-compose spec",
+    )
+    container_http_entry: Optional[str] = Field(
+        None,
+        alias="simcore.service.container-http-entrypoint",
+        description=(
+            "When a compose spec is provided, a container where the proxy "
+            "needs to send http traffic must be specified"
+        ),
+    )
+
+    needs_dynamic_sidecar: Optional[bool] = Field(
+        None, description="true if dynamic sidecar is needed to start that service"
+    )
+
+    @validator("needs_dynamic_sidecar", pre=True, always=True)
+    @classmethod
+    def auto_fill_dynamic_sidecar(cls, v, values):
+        if not v:
```

Comment:
what if a user sets ``needs_dynamic_sidecar==True`` and  ``paths_mapping==None``?

Reply:
As said by sandregg it's actually used more like a property and those values cannot be set by the user directly. As everything is derived from the labels.